### ISSUE TITLE
[Question]Using cf-java-client version 2 with oauth token

### ISSUE THREAD

**Issue by LeeU1911** [cloudfoundry/cf-java-client#517]
Hi Team,

I'm very excited with the new version of cf-java-client. Thanks for all the amazing work you have done so far.
I want to adopt this new version into my project but there is one thing that I don't know whether current version has supported for building cloudfoundryclient object using OAuth2 access token yet?

My application uses OAuth2 to authenticate user with UAA server, hence we don't keep username and password. I want to use the OAuth2 access token we have in Spring Security Context to build the CloudFoundryClient object and retrieve particular info per user.

Is that possible today?

Thanks again!

Best,
Vu

**Comment by nebhale** [comment]
At the moment, no we do not have the ability to configure the client in this way.  However, it is on our roadmap to implement before the 2.0.0 release.  The basic design is that you'll be able to contribute a "`TokenProvider`" implementation to the `CloudFoundryClient` builder which will be able to contribute the proper OAuth token as it choses.

**Comment by LeeU1911** [comment]
Hey Ben,

Thanks for your response! That sounds great although it means I can't use the new version as of today.

I'll come back when the support is ready. 

Best,
Vu

**Comment by nebhale** [comment]
We’re waiting on Spring Security to do a release that is Reactor-aware.  This is actively being worked on by @rwinch and once the support is there, we’ll have another look at it.

**Comment by jgrandja** [comment]
@nebhale @LeeU1911 The new OAuth2-OIDC client support we're building into Spring Security 5 automatically fetches the `UserInfo` resource from `UserInfo Endpoint` after the user authenticates and authorizes the client. Take a look at this [guide](https://github.com/spring-projects/spring-security/tree/master/samples/boot/oauth2login) to see what we have so far.

However, the current implementation does not provide a reactive implementation as of yet. @rwinch is working on the reactive support for Spring Security 5 and we're looking to provide reactive support for the OAuth2-OIDC client support soon.

**Comment by nebhale** [comment]
@jgrandja The hold up isn’t actually related to OAuth directly at all.  Instead, we’re waiting on Spring Security to put the `SecurityContextHolder` (or equivalent) into the Reactor Context (coming in Reactor 3.1).  That, in theory, will give us enough to propagate that token along with our requests.

**Comment by jgrandja** [comment]
@nebhale Ok thanks for the heads up.

**Comment by rwinch** [comment]
@nebhale Spring Security M4 which is planned for 14th September populates the Spring Security Authentication into the reactor `Context` within a WebFlux application. See https://github.com/spring-projects/spring-security/issues/4501

Is this what you are needing?

**Comment by nebhale** [comment]
I believe it is.  I’m not rushing to do this pre-release, as WebFlux isn’t available yet, but that’s the plan.

**Comment by rwinch** [comment]
@nebhale Ok thanks for the fast response. Please don't hesitate to reach out if you have anything you need from our side :)

**Comment by dmikusa-pivotal** [comment]
This issue is pretty old and a lot has changed since it was last discussed. I'm going to close this out. If you're still seeing something that you cannot do with the latest cf-java-client versions 4.x or 5.x then please open a new issue and let's discuss further. Thanks.
